### PR TITLE
Add OpenAPI generation and swagger to update-codegen.sh

### LINKED ISSUE #1271: API rule violations in Triggers types
# Expected Behavior

Triggers should have 0 API violations

# Actual Behavior

Running openapi-gen results in the following API violations:

```
 go run k8s.io/kube-openapi/cmd/openapi-gen \
    --input-dirs ./pkg/apis/triggers/v1beta1,knative.dev/pkg/apis,knative.dev/pkg/apis/duck/v1beta1 \
    --output-package ./pkg/apis/triggers/v1beta1 -o ./ \
    --go-header-file hack/boilerplate/boilerplate.go.txt
API rule violation: list_type_missing,github.com/tektoncd/triggers/pkg/apis/triggers/v1beta1,BitbucketInterceptor,EventTypes
API rule violation: list_type_missing,github.com/tektoncd/triggers/pkg/apis/triggers/v1beta1,CELInterceptor,Overlays
API rule violation: list_type_missing,github.com/tektoncd/triggers/pkg/apis/triggers/v1beta1,EventListenerSpec,TriggerGroups
API rule violation: list_type_missing,github.com/tektoncd/triggers/pkg/apis/triggers/v1beta1,EventListenerSpec,Triggers
API rule violation: list_type_missing,github.com/tektoncd/triggers/pkg/apis/triggers/v1beta1,EventListenerTrigger,Bindings
API rule violation: list_type_missing,github.com/tektoncd/triggers/pkg/apis/triggers/v1beta1,EventListenerTrigger,Interceptors
API rule violation: list_type_missing,github.com/tektoncd/triggers/pkg/apis/triggers/v1beta1,EventListenerTriggerGroup,Interceptors
API rule violation: list_type_missing,github.com/tektoncd/triggers/pkg/apis/triggers/v1beta1,GitHubInterceptor,EventTypes
API rule violation: list_type_missing,github.com/tektoncd/triggers/pkg/apis/triggers/v1beta1,GitLabInterceptor,EventTypes
API rule violation: list_type_missing,github.com/tektoncd/triggers/pkg/apis/triggers/v1beta1,NamespaceSelector,MatchNames
API rule violation: list_type_missing,github.com/tektoncd/triggers/pkg/apis/triggers/v1beta1,PodTemplate,Tolerations
API rule violation: list_type_missing,github.com/tektoncd/triggers/pkg/apis/triggers/v1beta1,TriggerBindingSpec,Params
API rule violation: list_type_missing,github.com/tektoncd/triggers/pkg/apis/triggers/v1beta1,TriggerInterceptor,Params
API rule violation: list_type_missing,github.com/tektoncd/triggers/pkg/apis/triggers/v1beta1,TriggerSpec,Bindings
API rule violation: list_type_missing,github.com/tektoncd/triggers/pkg/apis/triggers/v1beta1,TriggerSpec,Interceptors
API rule violation: list_type_missing,github.com/tektoncd/triggers/pkg/apis/triggers/v1beta1,TriggerTemplateSpec,Params
API rule violation: list_type_missing,github.com/tektoncd/triggers/pkg/apis/triggers/v1beta1,TriggerTemplateSpec,ResourceTemplates
API rule violation: list_type_missing,github.com/tektoncd/triggers/pkg/apis/triggers/v1beta1,WebhookInterceptor,Header
API rule violation: names_match,github.com/tektoncd/triggers/pkg/apis/triggers/v1beta1,EventListenerConfig,GeneratedResourceName
API rule violation: names_match,github.com/tektoncd/triggers/pkg/apis/triggers/v1beta1,InterceptorRequest,InterceptorParams
API rule violation: names_match,github.com/tektoncd/triggers/pkg/apis/triggers/v1beta1,KubernetesResource,WithPodSpec
API rule violation: names_match,github.com/tektoncd/triggers/pkg/apis/triggers/v1beta1,StatusError,s
API rule violation: names_match,github.com/tektoncd/triggers/pkg/apis/triggers/v1beta1,TriggerContext,EventID
API rule violation: names_match,github.com/tektoncd/triggers/pkg/apis/triggers/v1beta1,TriggerContext,EventURL
API rule violation: names_match,github.com/tektoncd/triggers/pkg/apis/triggers/v1beta1,TriggerContext,TriggerID
API rule violation: names_match,github.com/tektoncd/triggers/pkg/apis/triggers/v1beta1,TriggerSpecBinding,APIVersion
API rule violation: names_match,github.com/tektoncd/triggers/pkg/apis/triggers/v1beta1,TriggerSpecTemplate,APIVersion
API rule violation: names_match,github.com/tektoncd/triggers/pkg/apis/triggers/v1beta1,TriggerTemplateSpec,ResourceTemplates
```

# Steps to Reproduce the Problem

1.  ` go run k8s.io/kube-openapi/cmd/openapi-gen \
    --input-dirs ./pkg/apis/triggers/v1beta1,knative.dev/pkg/apis,knative.dev/pkg/apis/duck/v1beta1 \
    --output-package ./pkg/apis/triggers/v1beta1 -o ./ \
    --go-header-file hack/boilerplate/boilerplate.go.txt`

### PR DESCRIPTION
**Pull request by abayer**
# Changes

And fix #1271 along the way by using the same scripting as in https://github.com/tektoncd/pipeline/pull/4402 to ignore API rule violations for `name_match` which already exist. Any new mismatch will cause an error.

/kind misc

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [ ] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#tests) (if functionality changed/added)
- [ ] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#docs) (if user facing)
- [x] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commits)
- [x] Release notes block has been filled in or deleted (only if no user facing changes)

_See [the contribution guide](https://github.com/tektoncd/triggers/blob/master/CONTRIBUTING.md) for more details._

# Release Notes

```release-note
NONE
```

### CODE REVIEW

**Comment by abayer** [comment]
Oh, and I'll follow this up with a PR like https://github.com/tektoncd/pipeline/pull/4494 to add generated API reference docs.

File: hack/spec-gen/main.go [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+		},
+	}
+	jsonBytes, err := json.MarshalIndent(swagger, "", "  ")
+	if err != nil {
+		klog.Fatal(err.Error())
+	}
+	fmt.Println(string(jsonBytes))
+}
+
+func swaggify(name string) string {
```
Comment (dibyom) [inline hack/spec-gen/main.go:68]: question: Why do we need to this?

**Review by dibyom — APPROVED** [review]

**Comment by abayer** [comment]
fyi, once this merges, I'll have another PR that'll generate API reference docs like in https://github.com/tektoncd/pipeline/pull/4494 (though for Pipelines we currently have to use a fork of https://github.com/ahmetb/gen-crd-api-reference-docs because of the `pkg/apis/pipeline/pod` package's unversioned weirdness - here we don't need that).

File: hack/spec-gen/main.go (hunk 1 above)
Comment (dibyom) [reply hack/spec-gen/main.go:68]: ah ok. was curious as to why we need to have this custom function.

**Pull request merged**